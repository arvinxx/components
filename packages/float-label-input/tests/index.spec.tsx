import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import FloatLabelInput from '@arvinxu/float-label-input';

describe('FloatLabelInput', () => {
  it('正常渲染', () => {
    const { container } = render(<FloatLabelInput label={'test'} />);
    expect(container).toMatchSnapshot();
  });

  it('受控组件', () => {
    // const { container } = render(<FloatLabelInput label={'test'} />);
  });

  it('focus 向上移动; blur 返回', () => {
    const { queryByTestId } = render(<FloatLabelInput label={'test'} />);

    const input = queryByTestId('input');
    const label = queryByTestId('label');

    expect(label.style.transform).toBe('');
    fireEvent.focus(input);
    expect(label.style.transform).toBe('translateY(-28px)');
    fireEvent.blur(input);
    expect(label.style.transform).toBe('');
  });

  it('可输入外部高度', () => {
    // const { queryByTestId } = render(
    //   <FloatLabelInput label={'test'} style={{ height: 40 }} />,
    // );
    // const input = queryByTestId('input');
    // fireEvent.focus(input);
    // const label = queryByTestId('label');
    // debug(label);
    // expect(label.style.transform).toBe('translateY(-32px)');
  });
});
