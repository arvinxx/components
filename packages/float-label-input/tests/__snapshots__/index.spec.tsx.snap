// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatLabelInput 正常渲染 1`] = `
<div>
  <div
    class="avx-float-input-container"
    data-testid="container"
  >
    <input
      class="ant-input"
      data-testid="input"
      label="test"
      type="text"
      value=""
    />
    <label
      class="avx-float-input-label"
      data-testid="label"
      style="margin-left: 8px; margin-top: -26px;"
    >
      test
    </label>
  </div>
</div>
`;
